src/main/drivers/dma_reqmap.c

```diff
@@ -37,16 +37,149 @@
 typedef struct dmaPeripheralMapping_s {
     dmaPeripheral_e device;
     uint8_t index;
+#if defined(STM32H7)
+    uint8_t dmaRequest;
+#else
     dmaChannelSpec_t channelSpec[MAX_PERIPHERAL_DMA_OPTIONS];
+#endif
 } dmaPeripheralMapping_t;
 
 typedef struct dmaTimerMapping_s {
     TIM_TypeDef *tim;
     uint8_t channel;
+#if defined(STM32H7)
+    uint8_t dmaRequest;
+#else
     dmaChannelSpec_t channelSpec[MAX_TIMER_DMA_OPTIONS];
+#endif
 } dmaTimerMapping_t;
 
-#if defined(STM32F4) || defined(STM32F7)
+#if defined(STM32H7)
+
+#define REQMAP_SGL(periph) { DMA_PERIPH_ ## periph, 0, DMA_REQUEST_ ## periph }
```

Comment:
You mean the `REQMAP_TIM`.
I wasn't thinking it that far; just thought it would be nice to reuse the existing def for DMA request symbol macro (`DEF_TIM_DMA_REQ__BTCH_xxx`) would reduce code duplication.
I will replace it with an independent def for the `REQMAP_TIM`.

Reply:
I am all for deduplication. But I think the right way is to move the hard to read / debug notation that is in `timer_def.h` (and was a very useful to make defining custom targets easy) should be replaced with a list-based notation for the small number of Unified Targets that we'll be having.
In the past we've had a number of bugs in timer / DMA options lists, and they were always hard to spot / fix due to the complexity of the macro expansion that is involved.